packages/core/src/timeout/__tests__/timeout.test.ts

```diff
@@ -0,0 +1,117 @@
+import { describe, test, vi, expect } from 'vitest';
+import {
+  allSettled,
+  createEvent,
+  createStore,
+  createWatch,
+  fork,
+} from 'effector';
+
+import { createQuery } from '../../query/create_query';
+import { timeout } from '../timeout';
+import { isTimeoutError } from '../../errors/guards';
+
+describe('timeout(query, time)', () => {
+  test('timeout(query, number)', async () => {
+    const handler = vi.fn(
+      async () =>
+        new Promise((r) => {
+          setTimeout(r, 130);
+        })
+    );
+    const query = createQuery({
+      handler,
+    });
+
+    timeout(query, { after: 100 });
+
+    const scope = fork();
+
+    await allSettled(query.refresh, { scope, params: undefined });
+
+    expect(handler).toBeCalledTimes(1);
+    expect(isTimeoutError({ error: scope.getState(query.$error) })).toBe(true);
```

Comment:
🤓 I believe that it's better to check the whole error object, not only its type. For example, this test does not check that `timeouteError` includes a correct `timeout` field.

Reply:
Since these errors are part of separate internal errors library, i think, it should be covered by `src/errors/__tests__` instead 🤔 

I don't see any tests for error-creators themselves there, so will add them

Reply:
I meant that in this test, we do not sure that thrown error contains the same field `timeout` that we pass to `timeout operator`.

So, technically, we could introduce a bug like 👇

```ts
timout(query, { after: 10 })

query.finished.failed.watch(({ error ) => {
  errror.timeout // 100500
})
```

I think it's better to rewrite test with `.toEqual` checker
```ts
expect(scope.getState(query.$error)).toEqual(timeoutError({ timeout: 1000 });
```